Automattic/jetpack
# AI Client: remove obsolete blockListBlockWithAiDataProvider() HOC component

This janitorial PR cleans the blockListBlockWithAiDataProvider() HOC code since it isn't being used.

Fixes #

## Proposed changes:

* AI Client: remove obsolete blockListBlockWithAiDataProvider() HOC component

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Check the `blockListBlockWithAiDataProvider()` HOC is not imported/used for the Jetpack plugin
* Compile the app
* Go to the block editor
* Confirm there isn't any instance of `blockListBlockWithAiDataProvider`

<img width="518" alt="Screenshot 2023-10-21 at 09 31 31" src="https://github.com/Automattic/jetpack/assets/77539/07e237c2-adb5-4dfd-a12a-738372a4e971">

* Since it was used only by the Jetpack Form block, confirm it works as usual
